scripts/get_nn_models.py

```diff
@@ -0,0 +1,109 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Script for ingesting pretrained neural net models for this dataset.
+
+Running this script allows for updates or replacements of the existing model.
+
+Example:
+$ python get_nn_models.py -m rbResnet50-DC2
+imports rbResnet50-DC2 from /repo/main to this dataset's preloaded repo. See
+get_nn_models.py -h for more options.
+"""
+
+import argparse
+import logging
+import os
+import sys
+
+import lsst.log
+import lsst.skymap
+from lsst.daf.butler import Butler, CollectionType
+
+logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+lsst.log.configure_pylog_MDC("DEBUG", MDC_class=None)
+
+
+# Normal model usage does not use chains, but need them here both to decouple
+# this script from make_all.sh and to hook in to ap_verify's automated
+# pipeline execution.
+MODEL_PREFIX = "pretrained_models"  # Hardcoded into meas_transiNet, do not change
+MODEL_CHAIN = "models"
+
+# Avoid explicit references to dataset package to maximize portability.
+SCRIPT_DIR = os.path.abspath(os.path.dirname(__file__))
+DATASET_REPO = os.path.normpath(os.path.join(SCRIPT_DIR, "..", "preloaded"))
+
+
+########################################
+# Command-line options
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-b", dest="src_dir", default="/repo/main",
+                        help="Repo to import from, defaults to '/repo/main'.")
+    parser.add_argument("-m", dest="model_name", required=True,
+                        help="Model package to import.")
+    return parser
+
+
+args = _make_parser().parse_args()
+
+
+########################################
+# Clean up existing model
+
+def _clean_dataset(butler):
+    """Remove the previous model(s), to avoid clashes on load.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        A Butler pointing to the repository to be cleaned.
+    """
+    try:
+        model_runs = butler.registry.getCollectionChain(MODEL_CHAIN)
+    except lsst.daf.butler.registry.MissingCollectionError:
+        # No prior collections
+        return
+
+    butler.registry.setCollectionChain(MODEL_CHAIN, [])
+    butler.removeRuns(model_runs, unstore=True)
+
+
+dest = Butler(DATASET_REPO, writeable=True)
+_clean_dataset(dest)
+
+
+########################################
+# Transfer
+
+MODEL_COLLECT = f"{MODEL_PREFIX}/{args.model_name}"
```

Comment:
In the same spirit (of avoiding hard-coding): wouldn't it be better to use something like below?
https://github.com/lsst/meas_transiNet/blob/4daa84c48cf73e1126eb09e2947e85de51acd131/demo/ingest_demo.py#L11C1-L11C57

This way we utilize what is provided by the package itself, not needing to keep the code here up-to-date with potential future changes in meas_transiNet.

This may of course not match the philosophy of ap_verify. So I'm of course open to counter-arguments here.

Reply:
I assume that would look like:
```
package = NNModelPackage(args.model_name, 'butler', butler=src)
StorageAdapterButler.ingest(package, dest)
```

However, we'd still need to semi-automatically hook the collection into `defaults` for it to be reachable by `ap_verify` itself. So it seems to me that this script needs to know what the collection names are, even if no other part of `ap_verify` does.

Reply:
OK. I see how the butler->butler transfer is necessary here. Skip this suggestion and consider this point resolved.